Mlem/Logic/Participating/BlockPersonLogic.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  BlockPerson.swift
+//  Mlem
+//
+//  Created by Weston Hanners on 7/10/23.
+//
+
+import Foundation
+
+@MainActor
+func blockPerson(
+    account: SavedAccount,
+    person: APIPerson,
+    blocked: Bool) async throws -> Bool {
+        do {
+            let request = BlockPersonRequest(
+                account: account,
+                personId: person.id,
+                block: blocked
+            )
+            let response = try await APIClient().perform(request: request)
+            return response.blocked
+        } catch {
+            return false
```

Comment:
this function is declared `throws`, but we're catching all errors here and returning `false` so no throw will ever happen, it looks like the logic in `FeedPost` is written to show an error toast inside its `catch` but I don't think that will work if this function catches them all?

should we remove the `do/catch` from this function to let errors throw and be handled in the feed post?

Reply:
I'll look at adjusting the new contextual error thing to let us choose between alerts and toasts as the way to notify, so we can get the default behaviours without always showing alerts 👍